rewrite-java-core/src/test/java/TypeReferenceTest.java

```diff
@@ -0,0 +1,67 @@
+import com.azure.recipes.v2recipes.TypeReferenceRecipe;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+/**
+ * TypeReferenceTest is used to test out the recipe that changes the usage of TypeReference (azure core v1)
+ * to ParameterizedType (azure core v2)
+ * @author Ali Soltanian Fard Jahromi
+ */
+public class TypeReferenceTest implements RewriteTest {
+
+    /**
+     * This method sets which recipe should be used for testing
+     * @param spec stores settings for testing environment; e.g. which recipes to use for testing
+     */
+    @Override
+    public void defaults(RecipeSpec spec) {
```

Comment:
We should import all recipes to ensure we test how all recipes interact with each other. There might be some overlap that might cause some overwriting to happen and result in unexpected transformation. So, I would suggest having a base test class where all recipes are added in RecipeSpec and all other tests inheriting from it.

Reply:
Ensured all testing classes only use `rewrite.yml`. Moved all imperative (Java-based) recipes into `rewrite.yml`.

See changes in commit [492daeb](https://github.com/Azure/azsdk-java-rewrite-recipes/pull/14/commits/492daeb48360470829cc54218911b9b3e1758ace)